assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/applicationContext-spring-security.xml

```diff
@@ -253,19 +253,32 @@
 
   <bean id="sas" class="org.springframework.security.web.authentication.session.ChangeSessionIdAuthenticationStrategy" />
 
+  <bean id="loginAttemptService" class="org.pentaho.platform.engine.security.LoginAttemptService" scope="singleton">
+    <constructor-arg name="maxAttempt" value="10"></constructor-arg>
+    <constructor-arg name="cacheMinutes" value="1440"></constructor-arg>
+  </bean>
+
+  <bean id="preventBruteForceAuthenticationFailureListener"
+        class="org.pentaho.platform.web.http.security.PreventBruteForceAuthenticationFailureListener" >
+    <constructor-arg ref="loginAttemptService"/>
+  </bean>
+
   <bean id="authenticationProcessingFilter"
-        class="org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter">
+        class="org.pentaho.platform.web.http.security.PreventBruteForceUsernamePasswordAuthenticationFilter">
```

Comment:
Not sure it's good practice to have "PreventBruteForce" in the name of the classes, usually security sensitive information shouldn't stay so exposed as to be even mentioned in a configuration file.

Reply:
I understand your concern but the name of class should represent what class does. 

Beside that this configuration is on server side and is not exposed to the outside. This is open source code so even with a different name everyone could read the content and inffer what class does.